### ISSUE TITLE
OPX-acl-init service failed on z9100, s6010 platform after restart opx-create interface service

### ISSUE THREAD

**Issue by waliulislam** [open-switch/opx-nas-acl#22]
After installing PKGS_OPX-3.0.0-rc1-installer-x86_64.bin on z9100 and s6010 platform , I need to fanout few ports. I update that to /etc/opx/dn_nas_fanout_init_config.xml file. That required restart of opx-create-interface.service. After restart the service , opx-acl-init.service failed.

Here is steps to recreate the issue.
 
I did manual install from onie in my z9100 and reboot the box. Everything was working.

```
root@OPX:~# opx-show-version
OS_NAME="OPX"
OS_VERSION="unstable"
PLATFORM="Z9100-ON"
ARCHITECTURE="x86_64"
INTERNAL_BUILD_ID="OpenSwitch blueprint for Dell 1.0.0"
BUILD_VERSION="unstable.0"
BUILD_DATE="2018-08-15T16:33:10+0000"
INSTALL_DATE="2018-08-15T21:33:45+00:00"
SYSTEM_STATE= running
UPGRADED_PACKAGES=no
ALTERED_PACKAGES=yes
root@OPX:~# opx-show-system-status
System State:  running
No Failed Service
Modified Packages:
python-opx-snmp
root@OPX:~#
```

Then I change /etc/opx/dn_nas_fanout_init_config.xml file for all the ports that needed to be fanned out.

``` 
root@OPX:/mnt/onie-boot/admin# cat /etc/opx/dn_nas_fanout_init_config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!-- Copyright (c) 2015 Dell Inc.
Licensed under the Apache License, Version 2.0 (the "License"); you may
not use this file except in compliance with the License. You may obtain
a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 
THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
 
See the Apache Version 2.0 License for specific language governing
permissions and limitations under the License.
-->
<!--
    This file is used to store default init configuration for
    fanout interfaces
 
    example
 
<interfaces>
    <interface name="e101-001-0" fanout="4x1" speed="10G"/>
    <interface name="e101-002-0" fanout="1x1" speed="40G"/>
</interfaces>
 
-->
<interfaces>
    <interface name="e101-001-0" fanout="4x1" speed="10G"/>
    <interface name="e101-002-0" fanout="4x1" speed="10G"/>
    <interface name="e101-003-0" fanout="4x1" speed="10G"/>
    <interface name="e101-004-0" fanout="4x1" speed="10G"/>
    <interface name="e101-030-0" fanout="4x1" speed="10G"/>
</interfaces>
 
 
root@OPX:/mnt/onie-boot/admin#
``` 

That required a restart of opx-create-interface service. I did that.
 
```
root@OPX:/mnt/onie-boot/admin# systemctl restart opx-create-interface
root@OPX:/mnt/onie-boot/admin# opx-show-version
OS_NAME="OPX"
OS_VERSION="unstable"
PLATFORM="Z9100-ON"
ARCHITECTURE="x86_64"
INTERNAL_BUILD_ID="OpenSwitch blueprint for Dell 1.0.0"
BUILD_VERSION="unstable.0"
BUILD_DATE="2018-08-15T16:33:10+0000"
INSTALL_DATE="2018-08-15T21:33:45+00:00"
SYSTEM_UPTIME= 4 minutes
SYSTEM_STATE= degraded
UPGRADED_PACKAGES=no
ALTERED_PACKAGES=yes
root@OPX:/mnt/onie-boot/admin# opx-show-system-status
System State:  degraded
Failed Services
opx-acl-init.service
Modified Packages:
opx-nas-interface
python-opx-snmp
root@OPX:/mnt/onie-boot/admin# systemctl status opx-acl-init.service
? opx-acl-init.service - Default ACL entries for Control Plane Protocol packets
   Loaded: loaded (/lib/systemd/system/opx-acl-init.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2018-08-15 21:49:39 UTC; 2min 9s ago
  Process: 2431 ExecStart=/usr/bin/python -u /usr/bin/base_create_acl_entries.py (code=exited, status=1/FAILURE)
Main PID: 2431 (code=exited, status=1/FAILURE)
 
Aug 15 21:49:39 OPX systemd[1]: Started Default ACL entries for Control Plane Protocol packets.
Aug 15 21:49:39 OPX python[2431]: ('Inside ', 'INGRESS')
Aug 15 21:49:39 OPX python[2431]: [DSAPI:COMMIT], Failed to commit request at 0 out of 1
Aug 15 21:49:39 OPX python[2431]: Runtime Error: ACL INIT - Table creation failed: system-flow
Aug 15 21:49:39 OPX systemd[1]: opx-acl-init.service: Main process exited, code=exited, status=1/FAILURE
Aug 15 21:49:39 OPX systemd[1]: opx-acl-init.service: Unit entered failed state.
Aug 15 21:49:39 OPX systemd[1]: opx-acl-init.service: Failed with result 'exit-code'.
root@OPX:/mnt/onie-boot/admin# opx-ethtool e101-001-1
Settings for e101-001-1:
    Channel ID:   1
    Transceiver Status: Disable
    Media Type: QSFP 40GBASE CR4
    Part Number: 616750001
    Serial Number: CN05NP8R4BO7EGH
    Qualified: Yes
    Administrative State: DOWN
    Operational State: DOWN
    Supported Speed (in Mbps):  [10000]
    Auto Negotiation : off
    Configured Speed   : auto
    Operating Speed   : 0
    Duplex   : full
```

Workaround is to reboot the box after changing the fanout xml file rather than merely restarting the create-interface service.

**Comment by jeffyin-dell** [comment]
Being fixed by https://github.com/open-switch/opx-nas-interface/pull/46

**Comment by waliulislam** [comment]
Issue still exists on 3.0.0.0-rc2 build

**Comment by waliulislam** [comment]
@jeff-yin Can you please reopen the issue.

**Comment by jeffyin-dell** [comment]
@waliulislam I've reopened the issue.
I'm not sure if the above method is a supported method of doing fanout during switch run-time, even if it worked in the past. I know that these two scenarios are supported:

(1) Configure desired fanout settings in the `/etc/opx/dn_nas_fanout_init_config.xml` file and reboot the switch. The opx-create-interface service is a "one-shot" service that will fan-out the interfaces according to the configuration in the file.

(2) Use the `opx-config-fanout` utility during the switch run-time to fan-out/fan-in interfaces dynamically.

Anyway, I've reassigned @GarrickHe to take a look at this issue.

**Comment by GarrickHe** [comment]
@waliulislam @jeff-yin 
Restarting OPX services are not supported either individually or as a group. Otherwise we won't need to tell users to reboot after doing a 'apt-get dist-upgrade'. You'll actually need to restart the box.

```
Being fixed by open-switch/opx-nas-interface#46
```
#46 doesn't address this.

Thanks,
Garrick
